samples/rtos/threadx-demo/main.cpp

```diff
@@ -3,98 +3,75 @@
-static TX_BYTE_POOL tx_app_byte_pool;
-
-TX_THREAD thread_0;
-TX_THREAD thread_1;
-TX_THREAD thread_2;
-TX_THREAD thread_3;
-TX_QUEUE queue_0;
-TX_SEMAPHORE semaphore_0;
-TX_EVENT_FLAGS_GROUP event_flags_0;
-
-/// Function Prototypes
-void thread_0_entry(ULONG thread_input);
-void thread_1_entry(ULONG thread_input);
-void thread_2_entry(ULONG thread_input);
-void thread_3_entry(ULONG thread_input);
-
-// Merged tx_application_define and App_ThreadX_Init functions
-VOID tx_application_define(VOID* first_unused_memory) {
-    VOID* memory_ptr;
-    io::UART& uart = io::getUART<io::Pin::UART_TX, io::Pin::UART_RX>(9600);
-
-    if (tx_byte_pool_create(&tx_app_byte_pool, "Tx App memory pool", tx_byte_pool_buffer, TX_APP_MEM_POOL_SIZE)
-        != TX_SUCCESS) {
-        uart.printf("\n\rByte pool could not be created\n\r");
-    } else {
-
-        memory_ptr = (VOID*) &tx_app_byte_pool;
-
-        TX_BYTE_POOL* byte_pool = (TX_BYTE_POOL*) memory_ptr;
-
-        /* USER CODE BEGIN App_ThreadX_Init */
-        char* pointer = static_cast<char*>(TX_NULL);
-
-        /* Allocate the stack for thread 0.  */
-        tx_byte_allocate(byte_pool, (VOID**) &pointer, DEMO_STACK_SIZE, TX_NO_WAIT);
-
-        /* Create the main thread 0.  */
-        tx_thread_create(
-            &thread_0, "thread 0", thread_0_entry, 0, pointer, DEMO_STACK_SIZE, 1, 1, TX_NO_TIME_SLICE, TX_AUTO_START);
-
-        /* Allocate the stack for thread 1.  */
-        tx_byte_allocate(byte_pool, (VOID**) &pointer, DEMO_STACK_SIZE, TX_NO_WAIT);
-
-        /* Create the main thread 1.  */
-        tx_thread_create(
-            &thread_1, "thread 1", thread_1_entry, 0, pointer, DEMO_STACK_SIZE, 1, 1, TX_NO_TIME_SLICE, TX_AUTO_START);
-
-        /* Allocate the stack for thread 2.  */
-        tx_byte_allocate(byte_pool, (VOID**) &pointer, DEMO_STACK_SIZE, TX_NO_WAIT);
-
-        /* Create the main thread 2.  */
-        tx_thread_create(
-            &thread_2, "thread 2", thread_2_entry, 0, pointer, DEMO_STACK_SIZE, 1, 1, TX_NO_TIME_SLICE, TX_AUTO_START);
-
-        /* Allocate the stack for thread 3.  */
-        tx_byte_allocate(byte_pool, (VOID**) &pointer, DEMO_STACK_SIZE, TX_NO_WAIT);
-
-        /* Create the main thread 3.  */
-        tx_thread_create(
-            &thread_3, "thread 3", thread_3_entry, 0, pointer, DEMO_STACK_SIZE, 1, 1, TX_NO_TIME_SLICE, TX_AUTO_START);
+#define TX_APP_MEM_POOL_SIZE 65536
 
-        /* Create the message queue shared by all threads. */
-        tx_queue_create(&queue_0, "queue 0", TX_1_ULONG, pointer, DEMO_QUEUE_SIZE * sizeof(ULONG));
+/**
+ * Struct definition that holds all the counters for each thread.
+ */
+typedef struct counters {
+    uint32_t global_count; // Times a random number has been sent and received
+    uint32_t global_sum;   // Sum of random numbers
+    uint32_t thread0_count;
+    uint32_t thread0_sum;
+    uint32_t thread1_count;
+    uint32_t thread1_sum;
+    uint32_t thread2_count;
+    uint32_t thread2_sum;
```

Comment:
This would be better as an array of counters so you can have more reuse without having a massive if to update the thead 1 counts vs. thread 2.